### DISCUSSION TITLE
Updating `FooterKey` description doesn't work first time?

### DISCUSSION THREAD

**Discussion by TomJGooding** [Textualize/textual#4549] [Q&A] [answered]
I've been experimenting with updating the binding descriptions with the new `Footer` widget in v.0.63. This involves the `FooterKey` which I appreciate isn't part of the public API, but I'd like to understand why the update isn't working only on the first update.

```python
from textual.app import App, ComposeResult
from textual.widgets import Footer
from textual.widgets._footer import FooterKey


class DynamicBindingsApp(App):
    BINDINGS = [("space", "update_footer", "Count: 0")]

    count = 0

    def compose(self) -> ComposeResult:
        yield Footer()

    def action_update_footer(self) -> None:
        self.count += 1
        footer_key = self.query_one(FooterKey)
        footer_key.description = f"Count: {self.count}"
        footer_key.refresh()

        if self.count < 2:
            self.notify(
                "Footer description is not updated first time.",
                severity="error",
                timeout=1,
            )


if __name__ == "__main__":
    app = DynamicBindingsApp()
    app.run()
```

**Comment by willmcgugan** [comment, marked as answer]
I wouldn't expect this to work. There is no current blessed way of updating the description dynamically. Updating the description like that would be reset when the bindings changed with focus etc.

**Reply by TomJGooding** [reply]
Thanks Will for your reply. I completely understand there's no blessed way of updating the description dynamically, but I was curious to understand why the code above seemed to work apart from the first update. It looks like this actually _does_ now work with `v0.63.6` so I'll have a closer look at those changes rather than bothering you further.
